src/Tests/NodeIntegrationTest.php

```diff
@@ -255,4 +256,84 @@ public function testDataSet() {
     $this->assertNotNull($node->id(), 'Node ID is set, which means that the node has been auto-saved.');
   }
 
+  /**
+   * Tests that a required context value that is NULL throws an exception.
+   */
+  public function testRequiredNullValue() {
+    $entity_manager = $this->container->get('entity.manager');
+    $entity_manager->getStorage('node_type')
+      ->create(['type' => 'page'])
+      ->save();
+
+    $node = $entity_manager->getStorage('node')
+      ->create([
+        'title' => 'test',
+        'type' => 'page',
+      ]);
+
+    // Configure a simple rule with one action. The node ID is NULL and does not
+    // exist yet.
+    $action = $this->expressionManager->createInstance('rules_action',
```

Comment:
should use createAction() on the expression manager. Ideally, this wouldb extended to take ContextConfig as optional second argument imho, but that's another issue.